### PR TITLE
mavlink telemetry status only log simple HEARTBEAT validity

### PR DESCRIPTION
**Pull request by dagar**
@bkueng how about something like this? It's a bit verbose within mavlink receiver, but cheaper from the uORB perspective and stops any mavlink constants from leaking into the rest of PX4.

One downside is that there's an additional delay where mavlink has to first consider a heartbeat invalid (1.5 s timeout), then the commander timeout for data link loss, etc are from that point.

### CODE REVIEW

File: src/modules/mavlink/mavlink_main.h [hunk 1]
```diff
@@ -430,9 +430,7 @@ class Mavlink : public ModuleParams
 	/**
 	 * Get the receive status of this MAVLink link
 	 */
-	telemetry_status_s	&get_telemetry_status() { return _tstatus; }
-	void                    lock_telemetry_status() { pthread_mutex_lock(&_telemetry_status_mutex); }
-	void                    unlock_telemetry_status() { pthread_mutex_unlock(&_telemetry_status_mutex); }
```
Comment (bkueng) [inline src/modules/mavlink/mavlink_main.h:435]: `_telemetry_status_mutex` can be removed as well.

File: src/modules/mavlink/mavlink_receiver.h [hunk 2]
```diff
@@ -314,6 +315,26 @@ class MavlinkReceiver : public ModuleParams
 	// Allocated if needed.
 	TunePublisher *_tune_publisher{nullptr};
 
+	hrt_abstime _last_heartbeat_check{0};
+
+	hrt_abstime _heartbeat_type_antenna_tracker{0};
```
Comment (bkueng) [inline src/modules/mavlink/mavlink_receiver.h:320]: You could also store them as bits and clear all of them every 1-1.5s. The max delay for loss detection would be even higher though.

**Review by bkueng — DISMISSED** [review]
Thanks, this scales better.

File: src/modules/mavlink/mavlink_main.h (hunk 1 above)
Comment (dagar) [reply src/modules/mavlink/mavlink_main.h:435]: Done.

File: src/modules/mavlink/mavlink_receiver.h (hunk 2 above)
Comment (dagar) [reply src/modules/mavlink/mavlink_receiver.h:320]: I think that would be fine as well. We can strip this down to bits later if it's going to keep growing (more supported MAV_TYPEs/MAV_COMPONENTs).

**Pull request merged**